### PR TITLE
Delete tileset

### PR DESCRIPTION
**Pull request by mapsam**
Adds a `tilesets delete <id>` command for deleting a tileset. By default this requires an extra `y` prompt but you can pass `--force` to bypass.

cc @dianeschulze

### CODE REVIEW

File: mapbox_tilesets/scripts/cli.py [hunk 1]
```diff
@@ -197,6 +197,33 @@ def update(
...
+    if not force:
+        click.confirm(
+            "Are you sure you want to delete {0}?".format(tileset), abort=True
+        )
+
+    url = "{0}/tilesets/v1/{1}?access_token={2}".format(
+        mapbox_api, tileset, mapbox_token
+    )
+    r = requests.delete(url)
+    if r.status_code != 200 and r.status_code != 204:
```
Comment (dianeschulze) [inline mapbox_tilesets/scripts/cli.py:223]: Could you add some confirmation text here like we have for deleting a tileset source? e.g. 

```py
        click.echo("Tileset deleted.")
```

File: mapbox_tilesets/scripts/cli.py [hunk 2]
```diff
@@ -197,6 +197,33 @@ def update(
...
+
+    tilesets delete <tileset_id>
+    """
+
+    mapbox_api = _get_api()
+    mapbox_token = _get_token(token)
+
+    if not force:
+        click.confirm(
+            "Are you sure you want to delete {0}?".format(tileset), abort=True
```
Comment (dianeschulze) [inline mapbox_tilesets/scripts/cli.py:216]: Thoughts on prompting users to type the tileset name to add a little more friction?

File: tests/test_cli_delete.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+
+@pytest.mark.usefixtures("token_environ")
+@mock.patch("requests.delete")
+def test_cli_delete(mock_request_delete):
+    runner = CliRunner()
+
+    # sends expected request
+    mock_request_delete.return_value = MockResponse("", status_code=200)
+    result = runner.invoke(delete, ["test.id"], input="y")
```
Comment (dianeschulze) [inline tests/test_cli_delete.py:29]: Could you add a test where the input is `n` to the confirmation prompt?

File: mapbox_tilesets/scripts/cli.py (hunk 2 above)
Comment (dianeschulze) [reply mapbox_tilesets/scripts/cli.py:216]: Also, adding some warning text to the docs, e.g. "this action is not recoverable"

File: mapbox_tilesets/scripts/cli.py (hunk 1 above)
Comment (mapsam) [reply mapbox_tilesets/scripts/cli.py:223]: Yep I can do that - one thing we should consider is that the output of most of our commands are all JSON with a few exceptions. Feels odd that _some_ commands just output raw text. 

Something to consider for 2.x though

File: mapbox_tilesets/scripts/cli.py (hunk 2 above)
Comment (mapsam) [reply mapbox_tilesets/scripts/cli.py:216]: I can, should I do the same for the `delete-source` command?

File: tests/test_cli_delete.py (hunk 3 above)
Comment (mapsam) [reply tests/test_cli_delete.py:29]: Yep! Added in https://github.com/mapbox/tilesets-cli/pull/61/commits/4af26eb544a51a7f5b7df78e07a07e7e35dd7c68

File: mapbox_tilesets/scripts/cli.py (hunk 1 above)
Comment (mapsam) [reply mapbox_tilesets/scripts/cli.py]: `Tileset deleted.` added in https://github.com/mapbox/tilesets-cli/pull/61/commits/4af26eb544a51a7f5b7df78e07a07e7e35dd7c68!

File: mapbox_tilesets/scripts/cli.py (hunk 2 above)
Comment (dianeschulze) [reply mapbox_tilesets/scripts/cli.py:216]: Sure, makes sense to me!

**Comment by mapsam** [comment]
@dianeschulze updated the delete prompts to ask for the full ID - here's what it looks like:

tilesets
```shell
# error 
$ tilesets delete test.id
To confirm tileset deletion please enter the name of the "hello.world": hola.mundo
Error: hola.mundo does not match hello.world. Aborted!

# success
To confirm tileset deletion please enter the name of the tileset hello.world: hello.world
Tileset deleted.
```

sources
```shell
# error 
$ tilesets delete-source test waka
To confirm source deletion please enter the full source id "hello/world": hola/mundo
Error: hola/mundo does not match hello/world. Aborted!

# success
To confirm source deletion please enter the full source id "hello/world": hello/world
Source deleted.
```

**Pull request merged**